rate.go

```diff
@@ -1,28 +1,38 @@
 package throttled
 
 import (
+	"fmt"
 	"math"
 	"net/http"
 	"strconv"
 	"time"
+
+	"github.com/PuerkitoBio/throttled/store"
 )
 
 // Static check to ensure that rateLimiter implements Limiter.
 var _ Limiter = (*rateLimiter)(nil)
 
+const (
+	maxCASAttempts = 1024
```

Comment:
I guess the idea here is that even a single CAS conflict is quite unlikely given the speed of Go/Redis and the relative slowness of network connections, but it seems that if you ever had to come close to this upper bound of 1024, your rate limiter might be experiencing some pretty unexpected and serious performance degradation. Out of curiosity, is there any particular reason that 1024 is the magic number here?

Also, may be worth adding a small doc string. I personally didn't know the term "CAS" was until I read the code for a bit longer.